File: app/src/main/java/net/wojteksz128/worktimemeasureapp/notification/Channel.java
package net.wojteksz128.worktimemeasureapp.notification;

import android.app.NotificationChannel;
import android.content.Context;
import android.os.Build;
import android.support.annotation.RequiresApi;

import net.wojteksz128.worktimemeasureapp.R;

public enum Channel {

    END_OF_WORK_CHANNEL("end-of-work-channel", R.string.channel_end_of_work_name, R.string.channel_end_of_work_description, 4/*NotificationManager.IMPORTANCE_HIGH*/);

    private final String id;
    private final int name;
    private final int description;
    private final int importance;


    Channel(String id, int name, int description, int importance) {

        this.id = id;
        this.name = name;
        this.description = description;
        this.importance = importance;
    }

    @RequiresApi(api = Build.VERSION_CODES.O)
    public NotificationChannel getNotificationChannel(Context context) {
        String name = context.getString(this.getName());
        String description = context.getString(this.getDescription());

        final NotificationChannel notificationChannel = new NotificationChannel(id, name, importance);
        notificationChannel.setDescription(description);

        return notificationChannel;
    }

    public String getId() {
        return id;
    }

    public int getName() {
        return name;
    }

    public int getDescription() {
        return description;
    }

    @SuppressWarnings("unused")
    public int getImportance() {
        return importance;
    }
}

File: app/src/main/kotlin/net/wojteksz128/worktimemeasureapp/notification/Channel.kt
package net.wojteksz128.worktimemeasureapp.notification

import android.app.NotificationChannel
import android.content.Context
import android.graphics.Color
import android.os.Build
import androidx.annotation.RequiresApi
import androidx.annotation.StringRes
import androidx.core.app.NotificationManagerCompat

import net.wojteksz128.worktimemeasureapp.R

@Suppress("MemberVisibilityCanBePrivate")
enum class Channel(
    val id: String,
    @StringRes val channelName: Int,
    @StringRes val description: Int,
    val importance: Int,
) {
    END_WORK_TIME_CHANNEL(
        "end-work-time-channel",
        R.string.channel_end_work_time_name,
        R.string.channel_end_work_time_description,
        NotificationManagerCompat.IMPORTANCE_HIGH
    ),
    WORK_TIME_IN_PROGRESS_CHANNEL(
        "work-time-in-progress-channel",
        R.string.channel_work_time_in_progress_name,
        R.string.channel_work_time_in_progress_description,
        NotificationManagerCompat.IMPORTANCE_DEFAULT
    );

    @RequiresApi(api = Build.VERSION_CODES.O)
    fun getNotificationChannel(context: Context): NotificationChannel {
        val name = context.getString(this.channelName)
        val description = context.getString(this.description)

        val notificationChannel = NotificationChannel(id, name, importance)
        notificationChannel.description = description
        notificationChannel.enableLights(true)
        notificationChannel.lightColor = Color.BLUE

        return notificationChannel
    }
}
